cocos/2d/components/ui-opacity.ts

```diff
@@ -133,8 +134,32 @@ export class UIOpacity extends Component {
     @serializable
     protected _opacity = 255;
 
+    protected _parentAdded (node: Node): void {
+        const parent: Node |null = node.getParent();
+        if (parent == null) {
+            return;
+        }
+        const render = parent._uiProps.uiComp as UIRenderer;
+        if (render && render.color) {
+            return;
```

Comment:
Why return here?

Reply:
If it is a renderable element, the child nodes do not need to respond to changes.